emmet-core/emmet/core/search.py

```diff
@@ -261,3 +261,133 @@ class SearchDoc(PropertyDoc):
     # Theoretical
 
     theoretical: bool = Field(True, description="Whether the material is theoretical.")
+
+    @classmethod
+    def from_docs(cls, material_id: MPID, **docs: Dict[str, Dict]):
+        """Converts a bunch of search docs into a SearchDoc"""
+        doc = _copy_from_doc(docs)
+
+        # Reshape document for various sub-sections
+        # Electronic Structure + Bandstructure + DOS
+        if "bandstructure" in doc:
+            doc["has_props"].append("bandstructure")
+        if "dos" in doc:
+            doc["has_props"].append("dos")
+        if "calc_id" in doc:
+            doc["es_source_calc_id"] = doc["calc_id"]
+            del doc["calc_id"]
+
+        # Magnetism
+        doc["spin_polarized"] = "magnetism" in doc
+
+        doc["has_props"] = list(set(doc["has_props"]))
+
+        return SearchDoc(material_id=material_id, **doc)
+
+
+# Key mapping
+search_fields: Dict[str, list] = {
```

Comment:
Do we want to come up with some logic to pull fields from the document models instead of listing them here? Would be more robust to changes. Don't know what that might look like, but it might mean the search docs are also more verbose.

Reply:
Maybe in the future. For now, this should work fine.